python/lsst/pipe/tasks/matchBackgrounds.py

```diff
@@ -19,358 +19,473 @@
-                - ``isReference``: This is the reference exposure (only one
-                                   returned Struct will contain True for this
-                                   value, unless the ref exposure is listed multiple times).
-                - ``backgroundModel``: Differential background model
-                                       (afw.Math.Background or afw.Math.Approximate).
-                                       Add this to the science exposure to match the reference exposure.
-                - ``fitRMS``: The RMS of the fit. This is the sqrt(mean(residuals**2)).
-                - ``matchedMSE``: The MSE of the reference and matched images:
-                                  mean((refImage - matchedSciImage)**2);
-                    should be comparable to difference image's mean variance.
-                - ``diffImVar``: The mean variance of the difference image.
-                All fields except isReference will be None if isReference True or the fit failed.
+        result : `~lsst.afw.math.BackgroundList`
+            Differential background model
+            Add this to the science exposure to match the reference exposure.
 
         Raises
         ------
         RuntimeError
             Raised if an exposure does not exist on disk.
         """
-        numExp = len(expRefList)
-        if numExp < 1:
-            raise pipeBase.TaskError("No exposures to match")
-
-        if expDatasetType is None:
-            raise pipeBase.TaskError("Must specify expDatasetType")
-
-        if imageScalerList is None:
-            self.log.info("imageScalerList is None; no scaling will be performed")
-            imageScalerList = [None] * numExp
-
-        if len(expRefList) != len(imageScalerList):
-            raise RuntimeError("len(expRefList) = %s != %s = len(imageScalerList)" %
-                               (len(expRefList), len(imageScalerList)))
-
-        refInd = None
-        if refExpDataRef is None:
-            # select the best reference exposure from expRefList
-            refInd = self.selectRefExposure(
-                expRefList=expRefList,
-                imageScalerList=imageScalerList,
-                expDatasetType=expDatasetType,
-            )
-            refExpDataRef = expRefList[refInd]
-            refImageScaler = imageScalerList[refInd]
-
-        # refIndSet is the index of all exposures in expDataList that match the reference.
-        # It is used to avoid background-matching an exposure to itself. It is a list
-        # because it is possible (though unlikely) that expDataList will contain duplicates.
-        expKeyList = refExpDataRef.butlerSubset.butler.getKeys(expDatasetType)
-        refMatcher = DataRefMatcher(refExpDataRef.butlerSubset.butler, expDatasetType)
-        refIndSet = set(refMatcher.matchList(ref0=refExpDataRef, refList=expRefList))
+        if (numExp := len(warps)) < 1:
+            raise TaskError("No exposures to match")
```

Comment:
A production-ready task will need to work out some way to handle this case as well.

Reply:
Could just log a warning and return a struct with two empty lists?